cloud/aws/apigateway/monitors-api.tf

```diff
@@ -0,0 +1,113 @@
+resource "signalfx_detector" "heartbeat" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS ApiGateway heartbeat"
+
+	program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('IntegrationLatency', filter=filter('stat', 'mean') and filter('namespace', 'AWS/ApiGateway') and ${module.filter-tags.filter_custom})
+		not_reporting.detector(stream=signal, resource_identifier=['ApiName'], duration='${var.heartbeat_timeframe}').publish('CRIT')
+	EOF
+
+	rule {
+		description           = "has not reported in ${var.heartbeat_timeframe}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} on {{{dimensions}}}"
+	}
+}
+
+# Monitoring Api Gateway latency
+resource "signalfx_detector" "latency" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ApiGateway latency"
+
+	program_text = <<-EOF
+		from signalfx.detectors.aperiodic import aperiodic
+		signal = data('Latency', filter=filter('namespace', 'AWS/ApiGateway') and filter('stat', 'mean')and (not filter('Stage', '*'))and (not filter('Method', '*'))and (not filter('Resource', '*')) and ${module.filter-tags.filter_custom})${var.latency_aggregation_function}.${var.latency_transformation_function}(over='${var.latency_transformation_window}')
```

Comment:
may be a stat max ? I am not sure if this will be used while there is the "not filter" property.